client/landing/stepper/declarative-flow/hosted-site-migration-v2/index.tsx

```diff
@@ -11,14 +18,75 @@ export default {
 	},
 	isSignupFlow: false,
 	useSteps() {
-		return stepsWithRequiredLogin( [ STEPS.IMPORT_OR_MIGRATE_WITH_UPGRADE ] );
+		return stepsWithRequiredLogin( [
+			PLATFORM_IDENTIFICATION,
+			SITE_CREATION_STEP,
+			PROCESSING,
+			SITE_MIGRATION_UPGRADE_PLAN,
+		] );
 	},
 
-	//TEMP: This is a temporary implementation to make the flow work
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
 	useStepNavigation( currentStep, navigate ) {
+		const [ query ] = useSearchParams();
+
 		return {
-			submit: () => alert( 'Go' ),
+			submit: ( props?: ProvidedDependencies ) => {
+				const data = {
+					platform: props?.platform || query.get( 'platform' ),
+					siteId: props?.siteId || query.get( 'siteId' ),
+					siteSlug: props?.siteSlug || query.get( 'siteSlug' ),
+					next: props?.next || query.get( 'next' ),
+					url: props?.url,
+				} as Record< string, Primitive >;
+
+				if ( currentStep === PLATFORM_IDENTIFICATION.slug ) {
+					return navigate(
+						addQueryArgs(
+							{
+								platform: data.platform,
+								...( data.platform !== 'wordpress' ? { next: data.next } : {} ),
```

Comment:
I think it should be like this, correct?

```suggestion
								...( data.platform !== 'wordpress' ? { next: data.url } : {} ),
```

Reply:
Fixed here eb23a5841587a8afef7ed840626d890ec98ddb40
Thanks !